src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -0,0 +1,2375 @@
+        {
+            if (entered && !player->HasAura(SPELL_MEDITATION_TIMER_BAR))
+                player->CastSpell(player, SPELL_MEDITATION_TIMER_BAR, true);
+            else
+                player->RemoveAura(SPELL_MEDITATION_TIMER_BAR);
+
+            return true;
+        }
+        return false;
+    }
+};
+
+class spell_summon_troublemaker : public SpellScript
+{
+    PrepareSpellScript(spell_summon_troublemaker);
+
+    void HandleSummon(SpellEffIndex effIndex)
+    {
+        PreventHitDefaultEffect(effIndex);
+        uint32 entry = GetSpellInfo()->GetEffect(effIndex)->MiscValue;
+        SummonPropertiesEntry const* properties = sSummonPropertiesStore.LookupEntry(GetSpellInfo()->GetEffect(effIndex)->MiscValueB);
+        int32 duration = GetSpellInfo()->GetDuration();
+
+        if (!entry || !properties || !duration)
+            return;
+
+        Position pos = { 1181.75f, 3444.5f, 102.9385f, 3.285759f };
+        int32 radius = urand(0, 4) == 0 ? 0 : urand(1, 7);
+        float angle = M_PI + M_PI * (urand(0, 7) / 7.f);
+        pos.RelocateOffset({ radius * sin(angle), radius * cos(angle), 0.f, 0.f });
+
+        if (TempSummon* summon = GetCaster()->GetMap()->SummonCreature(entry, pos, properties, duration, GetCaster()))
+            summon->SetTempSummonType(TEMPSUMMON_CORPSE_TIMED_DESPAWN);
+    }
+
+    void Register() override
+    {
+        OnEffectHit += SpellEffectFn(spell_summon_troublemaker::HandleSummon, EFFECT_0, SPELL_EFFECT_SUMMON);
+    }
+};
+
+enum MeditationTimerSpells
+{
+    SPELL_CAVE_OF_SCROLLS_CREDIT            = 102447,
+    SPELL_CAVE_OF_SCROLLS_COMP_TIMER_AURA   = 128598
+};
+
+class spell_meditation_timer_bar : public AuraScript
+{
+    PrepareAuraScript(spell_meditation_timer_bar);
+
+    void HandleEffectPeriodic(AuraEffect const* /*aurEff*/)
+    {
+        if (Unit* target = GetTarget())
+        {
+            target->ModifyPower(POWER_ALTERNATE_POWER, 1);
+
+            if (target->GetPowerPct(POWER_ALTERNATE_POWER) == 100)
+            {
+                target->CastSpell(GetTarget(), SPELL_CAVE_OF_SCROLLS_CREDIT, true);
+                target->CastSpell(GetTarget(), SPELL_CAVE_OF_SCROLLS_COMP_TIMER_AURA, true);
+                target->RemoveAura(GetId());
+                target->ClearInCombat();
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_meditation_timer_bar::HandleEffectPeriodic, EFFECT_1, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+class spell_cave_of_scrolls_comp_timer_aura : public AuraScript
+{
+    PrepareAuraScript(spell_cave_of_scrolls_comp_timer_aura);
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Unit* target = GetTarget())
+            if (Creature* creature = target->FindNearestCreature(54567, target->GetVisibilityRange(), true))
```

Comment:
I could be wrong, but maybe use an enum like `NPC_AYSA_CLOUDSINGER` or something similar for creature 54567?

Reply:
That would be better for readability, yes.